RackHD/RackHD
# Updated rackHD installer scripts

This PR is to update the RackHD installer scripts as part of Phase 1 of RackHD Test Framework:
https://groups.google.com/forum/?hl=en#!topic/rackhd/cscQ2T20Uxs

Files updated are:
test/fit_tests/deploy/rackhd_source_install.py  - installs RackHD from source using rackhd_local.yml 
test/fit_tests/deploy/rackhd_package_install.py  - installs RackHD from packages using rackhd_package.yml 
test/fit_tests/config/global_config.json - contains repo and branch selections for source installer

The new installers are intended meet the following objectives:
- run either remote or local on management server (default)
- run from ESXi-based or Virtualbox-based management server VMs
- ability to run behind proxy
- run from Ubuntu 16 or Ubuntu 14 (see exception below)
- source installed from selectable repos and branches, default to RackHD/master
- install all RackHD package requirements, config files, network config
- start RackHD services without reboot
- check API to verify operational

The installers perform the following functions:
 - loads prerequisite packages git, ansible, etc.
 - downloads RackHD source or packages to management server 
 - installs using Ansible playbook
 - set up networking
 - load configuration files
 - startup and verify operations

Notes: 

Source installer fails on Ubuntu 14 due to a recent change in rackhd_local.yml than cause it to hang. If Ubuntu 14 is still supported then I will file this as a bug. Package installer works with both 14 and 16.

Source repos/branches installed via rackhd_source_install.py can be selected by changing the global_config.json file under the 'repos':'install' section. The configuration file format is likely to be changed when the Test Configuration Management story is completed. 

Updated source installer is currently running this Jenkins smoke test job in MN lab:
https://roebling.hwimo.lab.emc.com/jenkins/job/Running-Tests/job/RackHD-FIT-Smoke-Test/
